themes/client/html/default/catalog-detail.js

```diff
@@ -158,10 +158,14 @@ AimeosCatalogDetail = {
 	 */
 	selectVariant() {
 		const product = $('article.product');
-
+                const urlProductId = window.location.pathname.split('/').pop();
```

Comment:
Unfortunately, this wont't work in all environments as the variable may not be part of the path or has a prefix.
To solve that problem, we could move the PHP code to the modify() method which can then update a section in the cached HTML:
https://aimeos.org/docs/latest/frontend/html/create-subparts/#modify